modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2377,7 +2378,18 @@ def _arrow_type_to_dtype(cls, arrow_type):
         object
             Any dtype compatible with pandas.
         """
-        res = arrow_type.to_pandas_dtype()
+        import pyarrow
+
+        try:
+            res = arrow_type.to_pandas_dtype()
+        # Conversion to pandas is not implemented for some arrow types,
+        # perform manual conversion for them:
+        except NotImplementedError as e:
+            if pyarrow.types.is_time(arrow_type):
+                res = np.dtype(datetime.time)
+            else:
+                raise e
```

Comment:
```suggestion
                raise
```
this re-raises the exception keeping the original callstack as if no `except` clause was present, which makes it easier to understand things IMHO

Reply:
fixed